src/xepg.go

```diff
@@ -435,33 +439,19 @@ func createXEPGDatabase() (err error) {
 			m3uChannel.TvgName = m3uChannel.Name
 		}
 
-		Data.Cache.Streams.Active = append(Data.Cache.Streams.Active, m3uChannel.Name+m3uChannel.FileM3UID)
-
 		// Try to find the channel based on matching all known values.  If that fails, then move to full channel scan
 		m3uChannelHash := m3uChannel.TvgName + m3uChannel.FileM3UID
 		if m3uChannel.LiveEvent == "true" {
-			match := re.FindString(m3uChannel.Name)
-			if match != "" {
-				// Increment the counter for the group-title
-				numberLength := len(match)
-				groupCounters[m3uChannel.GroupTitle]++
-				lastNumberLength[m3uChannel.GroupTitle] = numberLength
-
-				// Create the new formatted name with the group-title and updated number
-				newName := fmt.Sprintf("%s%03d", m3uChannel.GroupTitle, groupCounters[m3uChannel.GroupTitle])
-				if len(match) == 2 {
-					newName = fmt.Sprintf("%s%02d", m3uChannel.GroupTitle, groupCounters[m3uChannel.GroupTitle])
-				} else if len(match) == 3 {
-					newName = fmt.Sprintf("%s%03d", m3uChannel.GroupTitle, groupCounters[m3uChannel.GroupTitle])
-				}
-				m3uChannelHash = newName
-			}
-
+			hash := md5.Sum([]byte(m3uChannel.URL + m3uChannel.FileM3UID))
+			m3uChannelHash = hex.EncodeToString(hash[:])
 		}
 
+		Data.Cache.Streams.Active = append(Data.Cache.Streams.Active, m3uChannelHash)
+
 		if val, ok := xepgChannelsValuesMap[m3uChannelHash]; ok {
 			channelExists = true
 			currentXEPGID = val.XEPG
+			currentChannelNumber = val.TvgChno
```

Comment:
@Fyb3roptik I _**think**_ this is what's causing my issue. Every time I try to change a channel number it hits this, which sets currentChannelNumber to the previous TvgChno. Then on line 498 it sets the channelID to currentChannelNumber.
What I don't understand: why aren't you running into this?